### PR TITLE
Serverless Image Similarity

### PR DESCRIPTION
**Pull request by omerXfaruq**

### CODE REVIEW

File: data/blog/2024-01-31-serverless-image-similarity.mdx [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+---
+
+
+[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/drive/1fis0Aj1PDeQFaBsKWoQp5CuKD2eP4EYD?usp=sharing)
+
+
+In this post, we'll guide you through the seamless creation of a face similarity system using Upstash Vector and the Huggingface Ecosystem, all within a serverless environment. With this powerful tech stack, you can eliminate concerns about the complexities of implementing and managing Backend, Frontend, Database, and Hosting. Instead, you can solely concentrate on the Machine Learning while we handle the intricacies of the Vector Database.
+
+
+This week, Upstash, a leading serverless data platform, unveiled its latest product - Vector. Utilizing Upstash Vector allows you to effortlessly to store and retrieve the most similar vectors based on your specified distance metric. Give it a try for free [here](https://upstash.com/)! For additional insights into Upstash Vector, explore our [blog](https://upstash.com/blog/introducing-vector-database).
```
Comment (melekup) [inline data/blog/2024-01-31-serverless-image-similarity.mdx:16]: effortlessly store

Comment (omerXfaruq) [inline data/blog/2024-01-31-serverless-image-similarity.mdx:16]:
```suggestion
This week, Upstash, a leading serverless data platform, unveiled its latest product - Vector. Utilizing Upstash Vector allows you to effortlessly store and retrieve the most similar vectors based on your specified distance metric. Give it a try for free [here](https://upstash.com/)! For additional insights into Upstash Vector, explore our [blog](https://upstash.com/blog/introducing-vector-database).
```

File: data/blog/2024-01-31-serverless-image-similarity.mdx [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+Let's install the necessary packages.
+```
+pip install transformers datasets upstash-vector gradio tqdm -q
+```
+
+Now it's time to initialize the model through transformers library. We're going to use a pretrained VIT from Google.
+```
+from transformers import AutoFeatureExtractor, AutoModel
+
+model_ckpt = "jayanta/vit-base-patch16-224-in21k-face-recognition"
```
Comment (omerXfaruq) [inline data/blog/2024-01-31-serverless-image-similarity.mdx:69]:
```suggestion
model_ckpt = "google/vit-base-patch16-224-in21k"
```

**Comment by enesakar** [comment]
closing the pull request as it is already published on huggingface:
https://huggingface.co/blog/omerXfaruq/serverless-image-similarity-with-upstash-vector

**Comment by omerXfaruq** [comment]
@enesakar IMO it would be nice to have it in our blog pages, it would make it richer and knowledge base would become bigger. Also it would also affect search engines.

**Pull request closed without merging**